Source/WebCore/dom/Element.cpp

```diff
@@ -4137,8 +4138,12 @@ void Element::updateFocusAppearance(SelectionRestorationMode, SelectionRevealMod
             return;
         
         // When focusing an editable element in an iframe, don't reset the selection if it already contains a selection.
-        if (this == frame->selection().selection().rootEditableElement())
+        if (this == frame->selection().selection().rootEditableElement()) {
+            FocusOptions options;
+            if (!options.preventScroll)
+                frame->selection().revealSelection();
```

Comment:
Wouldn't this just be equivalent to just calling `revealSelection()`?
```suggestion
            frame->selection().revealSelection();
```

Reply:
@whsieh - Done. It still progresses and I tried to